packages/driver/cypress/e2e/commands/querying/querying.cy.js

```diff
@@ -962,16 +1002,24 @@ describe('src/cy/commands/querying', () => {
         const button = cy.$$('#button').hide()
 
         cy.on('fail', (err) => {
-          const { lastLog } = this
+          assertLogLength(this.logs, 2)
 
-          expect(lastLog.get('state')).to.eq('failed')
-          expect(lastLog.get('error')).to.eq(err)
-          expect(lastLog.get('$el').get(0)).to.eq(button.get(0))
-          const consoleProps = lastLog.invoke('consoleProps')
+          const getLog = this.logs[0]
+          const assertionLog = this.logs[1]
+
+          expect(err.message).to.contain('This element `<button#button>` is not visible because it has CSS property: `display: none`')
+
+          expect(getLog.get('state')).to.eq('passed')
+          expect(getLog.get('error')).to.be.undefined
+          expect(getLog.get('$el').get(0)).to.eq(button.get(0))
+          const consoleProps = getLog.invoke('consoleProps')
 
           expect(consoleProps.Yielded).to.eq(button.get(0))
           expect(consoleProps.Elements).to.eq(button.length)
 
+          expect(assertionLog.get('state')).to.eq('failed')
+          expect(err.message).to.include(assertionLog.get('error').message)
+
```

Comment:
Get .get('button') is no longer marked as failed because it resolved correctly and the assertion was the failure.
|Before | After| 
| -- | -- |
|<img width="486" alt="Screen Shot 2022-11-02 at 7 15 42 PM" src="https://user-images.githubusercontent.com/14099737/199625754-fe8de3f1-cc3f-4e18-af9f-d7f2156813f6.png">|<img width="430" alt="Screen Shot 2022-11-02 at 7 18 00 PM" src="https://user-images.githubusercontent.com/14099737/199625900-e8a8d715-44f2-400c-8835-47ec2b39a6c8.png">|